addons/website/static/src/js/utils.js

```diff
@@ -239,6 +239,32 @@ function websiteDomain(self) {
     return ['|', ['website_id', '=', false], ['website_id', '=', websiteID]];
 }
 
+/**
+ * Checks if the 2 given URLs originates from the same domain.
+ * It will consider naked URL and `www` URL originating from the same domain.
+ * It will consider `https` URL `http` URL originating from the same domain.
+ *
+ * @param {string} url1
+ * @param {string} url2
+ * @returns {Boolean}
+ */
+function areSameDomain(url1, url2) {
```

Comment:
To be annoying, I would not name this using `areSameDomain` (`areSameDomains` or `areTheSameDomain` at least ^^) (same as some of your comments and variable names that may use domain where I would not use it). Feels wrong to have `areSameDomain('https://test.odoo.com', 'https://www.odoo.com') -> false` :grimacing: I thought about `areSimilarOrigins`("Similar" instead of "Same" to suggest that some "same" ones are considered in fact "similar" (https / http + www/naked)) but since you are in fact totally ignoring all protocol (ftp == http too), that may not be right either... in the end, this is very linked to the code you are fixing: do we need a named util ? If yes, `areCommonSelfRedirectingDomains` or `isHTTPSorNakedDomainRedirection` (to hint at the common redirection to itself with different protocol and subdomain (although you would need to check http/https explicitely too for this name to make sense :') ... I don't know :grimacing: